File: c++/constructor_2.cpp
#include <bits/stdc++.h>
using namespace std;

class Cat
{
  string name;
public:
  Cat(string);
  void cry() const;
};

Cat::Cat(string s) : name(s){}
void Cat::cry() const
{
  cout << "にゃあ。俺様は" << name << "だ。" << endl;
}

int main()
{
  string st;
  cout << "猫を生成します。名前を入力してください。" << endl;
  cin >> st;

  Cat dora(st);
  cout << "あなたの名づけた猫がメモリ上に生成されました。" << endl;
  cout << "猫が鳴きます。" << endl;

  dora.cry();
}

File: ruby/constructor_2.rb
class Cat
  def initialize(s)
    @name = s
  end

  def cry
    puts "にゃあ。俺様は#{@name}だ。"
  end
end

puts "猫を生成します。名前を入力してください。"
st = gets.chomp

dora = Cat.new(st)
puts "あなたの名づけた猫がメモリ上に生成されました。"
puts "猫が鳴きます。"
dora.cry
